lib/fog/dynect/models/dns/record.rb

```diff
@@ -25,7 +25,9 @@ def save(replace=false)
           }
           options.delete_if {|key, value| value.nil?}
 
-          if replace
+          refresh_id if replace && identity.nil?
+          if replace && !identity.nil?
```

Comment:
The else block at line 32 is executed under different conditions than refresh_id, so it would have to be something like:

```
if replace
  if identity.nil?
    refresh_id
    data = service.post_record(type, zone.identity, name, rdata, options).body['data']
  else
    options['record_id'] = identity
    data = service.put_record(type, zone.identity, name, rdata, options).body['data']
  end
else
  data = service.post_record(type, zone.identity, name, rdata, options).body['data']
end
```

If you think that is more clear, then I can change it, but I really did not want to duplicate the post_record call.